server/src/main/java/com/example/service/UploadService.java

```
@@ -24,21 +23,22 @@
 	private ApplicationContext applicationContext;
 	private TaskExecutor taskExecutor;
 
-	public UploadLog uploadFileToProfile(String profileId, File file) {
-		System.out.println(profileId);
-		System.out.println(file.getName());
+	public UploadLog uploadFileToProfile(List<String> profileIds, File file) {
+		Set<Machine> machines = new HashSet<>();
 
-		Optional<Profile> optionalProfile = profileRepository.findById(profileId);
-		if (optionalProfile.isPresent()) {
-			Profile profile = optionalProfile.get();
-			Set<Machine> machines = profile.getMachines();
-			UploadLog uploadLog = uploadLogService.save(profile, file.getName(), machines.size());
-
-			taskExecutor.execute(createFileUploadTask(uploadLog, file, machines));
+		for (String profileId : profileIds) {
+			Profile profile = profileRepository.findProfileById(profileId);
+			machines.addAll(profile.getMachines());
+		}
 
-			return uploadLog;
+		if (machines.isEmpty()) {
+			return null;
```

probably you want to throw an exception here

---

discuss on monday :)